spec/integration/volumes_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'spec_helper'
+require 'time'
+require 'fileutils'
+
+describe "chef-provisioning-docker" do
+  extend DockerSupport
+  include DockerConfig
+
+  when_the_chef_12_server "exists", organization: "foo", server_scope: :context, port: 8900..9000 do
+    with_docker "volumes supported" do
+
+      # owing to how RSpec works, things defined by let() are not accessible in the recipes we define inside
+      # expect_converge{}.
+      ubuntu_options = {
+          :base_image => {
+            :name => 'ubuntu',
+            :repository => 'ubuntu',
+            :tag => '14.04'
+          }
+        }
+
+      let(:timestamp) { Time.now.to_i }
+      docker_driver = driver
+
+      context "machine_image resource" do
+
+        let(:spec_image_tag) { "docker_image_spec_#{timestamp}" }
+
+        after(:each) {
+          image = docker_driver.find_image("chef:#{spec_image_tag}")
+          image.delete(force: true) unless image.nil?
+          FileUtils.rm_rf("/tmp/opt")
+          FileUtils.rm_rf("/tmp/#{spec_image_tag}")
+        }
+
+        it "installs chef omnibus on the volume" do
+          tag = spec_image_tag
+
+          expect_converge {
+
+            machine_image tag do
+              machine_options :docker_options => ubuntu_options
+              action :create
+            end
+          }.not_to raise_error
+
+          expect(File.exists?("/tmp/opt/chef/bin/chef-client")).to be_truthy
```

Comment:
It mounts it following way by default:
```ruby
@chef_volume = (convergence_options[:chef_volume] || '/tmp/opt/chef') + ":/opt/chef"
```
On you host it should be `/tmp/opt/chef` and in container `/opt/chef`.
This can be overridden with (see code below): 
```
:convergence_options => {
  :chef_volume => "/tmp/#{tag}/chef"
}
```